src/libsystemd/sd-id128/id128-util.c

```diff
@@ -102,16 +109,17 @@ int id128_read_fd(int fd, Id128FormatFlag f, sd_id128_t *ret) {
         return r == -EINVAL ? -EUCLEAN : r;
 }
 
-int id128_read(const char *root, const char *p, Id128FormatFlag f, sd_id128_t *ret) {
-        _cleanup_close_ int fd = -EBADF;
+int id128_read(const char *root, const char *path, Id128FormatFlag f, sd_id128_t *ret) {
+        _cleanup_free_ char *filename = NULL;
+        _cleanup_close_ int dir_fd = -EBADF;
 
-        assert(p);
+        assert(path);
 
-        fd = chase_and_open(p, root, CHASE_PREFIX_ROOT, O_RDONLY|O_CLOEXEC|O_NOCTTY, /* ret_path = */ NULL);
-        if (fd < 0)
-                return fd;
+        dir_fd = chase_and_open_parent(path, root, CHASE_PREFIX_ROOT, &filename);
+        if (dir_fd < 0)
+                return dir_fd;
```

Comment:
Why open the parent here? We can open the file itself and use `id128_read_fd()` here.

I'd use a regular `chase()` here to get the file descriptor since `id128_read_at()` will reopen the fd for us anyway.